app/client/cli/servicer.go

```diff
@@ -115,13 +119,43 @@ func validateServicer(session *rpc.Session, servicerAddress string) (*rpc.Protoc
 	return nil, fmt.Errorf("Error getting servicer: address %s does not match any servicers in the session %d", servicerAddress, session.SessionNumber)
 }
 
+// ADDTEST: add either unit tests or integration tests using kvstore
+// getSessionFrom uses the client-side session cache to retrieve a session for app/chain combination at the provided height, if one has already been retrieved and cached.
+func getSessionFromCache(sessionCache *cache.SessionCache, appAddress, chain string, height int64) (*rpc.Session, error) {
+	if sessionCache == nil {
+		return nil, fmt.Errorf("got nil session cache")
+	}
+
+	session, err := sessionCache.Get(appAddress, chain)
+	if err != nil {
+		return nil, err
+	}
+
+	// verify the cached session is valid
+	if height >= session.SessionHeight && height < session.SessionHeight+session.NumSessionBlocks {
+		return session, nil
+	}
+
+	return nil, fmt.Errorf("no valid session found")
+}
+
 func getCurrentSession(ctx context.Context, appAddress, chain string) (*rpc.Session, error) {
 	// CONSIDERATION: passing 0 as the height value to get the current session seems more optimal than this.
 	currentHeight, err := getCurrentHeight(ctx)
 	if err != nil {
 		return nil, fmt.Errorf("Error getting current session: %w", err)
 	}
 
+	sessionCache, err := cache.NewSessionCache(sessionCacheDBPath)
```

Comment:
Wdyt of moving the initialization into an `init` function?

Reply:
Thank you for the review. Fixed.